resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/Utils.scala

```diff
@@ -153,22 +153,27 @@ object Utils extends Logging {
   }
 
   def createTarGzFile(inFile: String, outFile: String): Unit = {
+    val oFile = new File(outFile)
     val fileToTarGz = new File(inFile)
     Utils.tryWithResource(
       new FileInputStream(fileToTarGz)
     ) { fis =>
       Utils.tryWithResource(
         new TarArchiveOutputStream(
           new GzipCompressorOutputStream(
-            new FileOutputStream(
-              new File(outFile))))
+            new FileOutputStream(oFile)))
       ) { tOut =>
         val tarEntry = new TarArchiveEntry(fileToTarGz, fileToTarGz.getName)
+        // Each entry does not keep the file permission from the input file.
+        // Setting permissions in the input file do not work. Just simply set
+        // to 777.
+        tarEntry.setMode(0x81ff)
```

Comment:
This looks like another orthogonal issue to me. Could you spin-off this one?

Reply:
Its slightly orthogonal but related to the current PR. The custom python in the tests should have executable permission and tgz should keep the permission. Without this change, the test fails because it loses executable permission and cannot be used as a python alone later. FWIW this util is for test-only, and this PR adds the first case in the test where it needs to keep the file permissions.